src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/Orchestrator/DateTimeAnalyzer.cs

```diff
@@ -20,43 +20,36 @@ public class DateTimeAnalyzer : DiagnosticAnalyzer
         private const string Category = SupportedCategories.Orchestrator;
         public const DiagnosticSeverity severity = DiagnosticSeverity.Warning;
 
-        private static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
+        public static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
-
-        public override void Initialize(AnalysisContext context)
+        public static bool RegisterDiagnostic(SyntaxNode method, CompilationAnalysisContext context, SemanticModel semanticModel)
         {
-            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
-            context.EnableConcurrentExecution();
-            context.RegisterSyntaxNodeAction(AnalyzeIdentifierDateTimeNow, SyntaxKind.IdentifierName);
-        }
+            var diagnosedIssue = false;
 
-        private static void AnalyzeIdentifierDateTimeNow(SyntaxNodeAnalysisContext context)
-        {
-            var identifierName = context.Node as IdentifierNameSyntax;
-            if (identifierName != null)
+            foreach (SyntaxNode descendant in method.DescendantNodes())
             {
-                var identifierText = identifierName.Identifier.ValueText;
-                if (identifierText == "Now" || identifierText == "UtcNow" || identifierText == "Today")
+                var identifierName = descendant as IdentifierNameSyntax;
+                if (identifierName != null)
```

Comment:
Tip: you can save a line of code by doing this instead if doing a separate `as` operation + `null` check:
```csharp
if (descendant is IdentifierNameSyntax identifierName)
{
    ...
}
```

I see other analyzers use the same pattern.

Reply:
I suppose that makes sense if I don't have to cast it many times throughout my methods, which I don't think I'd need to do for these.

Reply:
Oh cool, I didn't realize I could do it this way.